### DISCUSSION TITLE
How do declare an auto-generated column?

### DISCUSSION THREAD

**Discussion by massiccio** [tortoise/tortoise-orm#1394] [General]
Hi,

I am using sqlite and would like to have a few columns being generated from a json field.
According to https://tortoise.github.io/fields.html#tortoise.fields.base.Field.allows_generated I should set allows_generated=True and set GENERATED_SQL, but I could not find any example.

This works fine in the sqlite3 console
```
CREATE TABLE "alert" (
    "id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
    "tags" JSON NOT NULL,
    "team" VARCHAR(20) as (json_extract(tags, '$.team')),
);
```

However the following model ignores the "team" column

```
class Alert(models.Model):
    tags = fields.JSONField(validators=[TagsValidator()])
    team = fields.CharField(max_length=20, index=True, null=False, generated=True, allows_generated=True)
    team.GENERATED_SQL = """'team' VARCHAR(20) as (json_extract(tags, '$.team'))"""
```

I have also tried to assign "as (json_extract(tags, '$.team'))" similar to GENERATED_SQL, however that field is always NULL.
Any idea?

sqlite documentation: https://www.sqlite.org/gencol.html

Thank you.

**Comment by Teut2711** [comment]
No, you are understanding it wrong. 
If you inherit a field type and create a new field type, then you would use GENERATED_SQL. You can see none has that other than IntField

Check here 

https://tortoise.github.io/_modules/tortoise/fields/data.html#IntField
